tests/components/dwd_weather_warnings/test_config_flow.py

```diff
@@ -0,0 +1,157 @@
+        return_value=True,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=DEMO_CONFIG_ENTRY
+        )
+
+        assert result["type"] == FlowResultType.CREATE_ENTRY
+        assert result["title"] == EXPECTED_NAME
+        assert result["data"] == {
+            CONF_NAME: EXPECTED_NAME,
+            CONF_REGION_IDENTIFIER: DEMO_CONFIG_ENTRY[CONF_REGION_IDENTIFIER],
+        }
+
+
+async def test_create_entry_invalid_region_identifier(hass: HomeAssistant) -> None:
+    """Test that the user step fails with a wrong region identifier."""
+    with patch(
+        "homeassistant.components.dwd_weather_warnings.config_flow.DwdWeatherWarningsAPI",
+        return_value=False,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=DEMO_CONFIG_ENTRY
+        )
+
+        assert result["errors"] == {"base": "invalid_identifier"}
+
+
+async def test_import_flow_full_data(hass: HomeAssistant) -> None:
+    """Test a successful import of a full YAML configuration."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_IMPORT}, data=DEMO_YAML_CONFIGURATION.copy()
+    )
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["title"] == DEMO_YAML_CONFIGURATION[CONF_NAME]
+
+    result_data = DEMO_YAML_CONFIGURATION.copy()
+    result_data[CONF_REGION_IDENTIFIER] = result_data.pop(CONF_REGION_NAME)
+
+    assert result["data"] == result_data
+
+
+async def test_import_flow_no_name(hass: HomeAssistant) -> None:
+    """Test a successful import of a YAML configuration with no name set."""
+    data = DEMO_YAML_CONFIGURATION.copy()
+    data.pop(CONF_NAME)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_IMPORT}, data=data
+    )
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["title"] == EXPECTED_NAME
+
+    result_data = DEMO_YAML_CONFIGURATION.copy()
+    result_data[CONF_NAME] = EXPECTED_NAME
+    result_data[CONF_REGION_IDENTIFIER] = result_data.pop(CONF_REGION_NAME)
+
+    assert result["data"] == result_data
+
+
+async def test_import_flow_only_required(hass: HomeAssistant) -> None:
+    """Test a successful import of a YAML configuration with only required properties."""
+    data = DEMO_YAML_CONFIGURATION.copy()
+    data.pop(CONF_NAME)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_IMPORT}, data=data
+    )
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["title"] == EXPECTED_NAME
+
+    result_data = DEMO_YAML_CONFIGURATION.copy()
+    result_data[CONF_NAME] = EXPECTED_NAME
+    result_data[CONF_REGION_IDENTIFIER] = result_data.pop(CONF_REGION_NAME)
+
+    assert result["data"] == result_data
```

Comment:
How is that different to the test above?

Reply:
It wasn't, I forgot to remove the monitored conditions from the copied dict 🥲